xbmc/cores/AudioEngine/Sinks/pipewire/AESinkPipewire.cpp

```diff
@@ -460,6 +462,17 @@ unsigned int CAESinkPipewire::AddPackets(uint8_t** data, unsigned int frames, un
 
   stream->QueueBuffer(pwBuffer);
 
+  const auto end = std::chrono::steady_clock::now();
+
+  m_latency = std::chrono::duration<double, std::ratio<1>>(static_cast<double>(frames) /
+                                                           m_format.m_sampleRate);
+
+  int ret = loop->Wait(m_latency - (end - start));
+  if (ret < 0 && ret != -ETIMEDOUT)
+  {
+    CLog::Log(LOGERROR, "CAESinkPipewire::{} - wait error: {}", __FUNCTION__, strerror(ret));
+  }
+
```

Comment:
Please only block when your buffer is full, else you will underrun when starting playback or after seek. You could track your delay from GetDelay.

Reply:
```
Adds packets to be sent out, this routine MUST block or sleep.
```
If you add 100 ms and sleep 100 ms -> your sink is dry after that again. Therefore most sinks block by themselves, but only if their buffer is empty.

Something like:
while(space in buffer less than period) block();
write into buffer()
return